src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -537,6 +538,10 @@ QgsOgrProvider::QgsOgrProvider( QString const &uri, const ProviderOptions &optio
   setNativeTypes( nativeTypes );
 
   QgsOgrConnPool::instance()->ref( QgsOgrProviderUtils::connectionPoolId( dataSourceUri( true ), mShareSameDatasetAmongLayers ) );
+
+  connect( this, &QgsOgrProvider::aboutToCalculateExtent, QgsProject::instance(), &QgsProject::initCalculateExtentProgress );
```

Comment:
This logic is backward -- a project would need to connect to the data providers from the layers which are added to the project and not vice-versa. (Specifically, we **can't** access QgsProject::instance() from a data provider or map layer, as the layer may not belong to QgsProject::instance()).

Reply:
> What really keeps confusing me for days now is **how to get a dataprovider's mapLayer** which it somehow should to be connected to. No idea if this not implemented yet, was I simply unable to find it or is there a logical gap that I am having..

I found out that a dataProvider may serve several mapLayers which means that different QgsMapLayer objects may relate to the same QgsDataProvider object.

That information is not crucial and only relevant for corner cases like having several large filtered layers in a project that will not receive a correct display of the progress bar.

Currently I have no idea where to transmit this information from and how. This is the debugger state when getting to the `QgsOgrProvider::extent()` calculation function:
![image](https://user-images.githubusercontent.com/67278094/163282459-b2162198-0790-4047-bc7b-fe0efe6e04f5.png)